test/coverage/datamodel/UuidSerializerTest.cpp

```diff
@@ -23,16 +27,159 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUuidSerializer() = default;
+	~TestUuidSerializer() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
+// Test string serialization
+TEST_F(TestUuidSerializer, SerializeToString) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABCDEF);
+	uuid.set_lsb(0xFEDCBA0987654321);
+
+	std::string uuid_str =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	EXPECT_EQ(uuid_str, "1234567890abcdef-fedcba0987654321");
+}
+
+// Test string deserialization
+TEST_F(TestUuidSerializer, DeserializeFromString) {
+	std::string uuid_str = "1234567890abcdef-fedcba0987654321";
+	uprotocol::v1::UUID uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::deserialize(uuid_str);
+
+	EXPECT_EQ(uuid.msb(), 0x1234567890ABCDEF);
+	EXPECT_EQ(uuid.lsb(), 0xFEDCBA0987654321);
```

Comment:
It wouldn't hurt to test a couple more values, such as one with mixed case letters and one with some leading zeros.